pkg/lib/saml/service.go

```diff
@@ -137,6 +147,7 @@ func (s *Service) IdpMetadata(serviceProviderId string) (*samlprotocol.Metadata,
 					SingleLogoutServices: sloServices,
 				},
 				SingleSignOnServices: ssoServices,
+				Attributes:           attributes,
```

Comment:
I did not read the spec. But is there any other XML attribute or XML element to tell whether a attribute is always present?

Reply:
For attribute,  do you mean SAML attribute, or attribute in an XML element?
If SAML attribute, I think the metadata is telling what attribute this idp supports. However, it is quite common that the metadata does not include such information or it does not match with the actual output in other providers.